### DISCUSSION TITLE
Aborting supabase request in supabasejs

### DISCUSSION THREAD

**Discussion by kaung-minkhant** [supabase/supabase#27340] [Questions] [answered]
Hii, I am using supabase for my work and I have having some problem aborting the supabase request. I love in a relatively slow connections area and it can be so slow that the request just timeout. That is what we want but the timeout is too long and we want to attach abort signal to all the supabase request. I have figured out how to do this with select queries with abortSignal but can't figure out how for auth.

If you have any idea or resources please let me know. Thanks in advanced.

**Comment by silentworks** [comment, marked as answer]
From reading the docs this doesn't seem possible from the auth flow perspective, but since you can pass your own `fetch` handler to the supabase-js library you should be able to use the WebAPIs abort controller in any case https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort.

**Reply by kaung-minkhant** [reply]
Can you please point me how to use my own fetch? Probably with an example. This is very new to me, going through the docs and source I only thought I cannot give my own fetch. I wanted to use Abort controller but since I do not know how to introduce it into supabase client.

**Reply by kaung-minkhant** [reply]
Ok, I found how to use my own fetch from the docs, but still can you please provide me with an example using abort controller? Much appreciated 😊

**Reply by silentworks** [reply]
The link I posted in the original post has an example of using the AbortController, you can follow the code there and implement in your own fetch handler.

**Reply by kaung-minkhant** [reply]
Hiii, thanks for your answer, it is indeed working to abort it. I have a follow up question for this. Is it possible to actually close the connection. I mean, due to the slow connection, the connection is aborted but, the request still went through and added an entry to database after timeout. I set the timeout to like 10s, but the connection is so slow that is take more than 10s to complete, but i have aborted the fetch request. How should I handle the situations, is there any recommended way to handle this? Very much thanks in advance.
